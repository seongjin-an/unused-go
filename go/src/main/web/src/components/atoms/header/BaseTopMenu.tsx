import React, { MouseEvent } from 'react';
import styled from 'styled-components';

interface IProps {}

export const BaseTopMenu: React.FC<IProps> = ({}) => {
    return (
        <StyledTopMenus>
            <li>나의 중고시새</li>
            <li>운동용품</li>
            <li>의류</li>
            <li>커뮤니티</li>
        </StyledTopMenus>
    );
};

const StyledTopMenus = styled.ul`
    margin: 0;
    padding: 0;
    display: flex;
    webkit-box-align: center;
    align-items: center;
    width: auto;
    height: 42px;
    cursor: pointer;
    list-style: none;

    & > li {
        font-family: PretendardBold;
        font-size: 18px;
        color: #f6f6f6;
    }

    //& > li:not(:last-child){
    //  margin-right: 105px;
    //}
    & > li:first-of-type {
        margin-right: 104px;
    }
    & > li:nth-of-type(2) {
        margin-right: 100px;
    }
    & > li:nth-of-type(3) {
        margin-right: 121px;
    }
`;
