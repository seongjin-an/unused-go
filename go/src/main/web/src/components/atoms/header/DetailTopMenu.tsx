import React, { MouseEvent } from 'react';
import styled, { css, keyframes } from 'styled-components';

interface IProps {
    isShow: boolean;
}

export const DetailTopMenu: React.FC<IProps> = ({ isShow }) => {
    return (
        <StyledDetailMenu isShow={isShow}>
            <ul>
                <li>
                    <ul>
                        <li>중고시세 한눈에보기</li>
                    </ul>
                </li>
                <li>
                    <ul>
                        <li>운동용품 전체</li>
                        <li>아령</li>
                        <li>바벨</li>
                        <li>원판/플레이트</li>
                        <li>벨트</li>
                        <li>악력기</li>
                        <li>밴드/짐볼</li>
                        <li>스트랩/보호대</li>
                    </ul>
                </li>
                <li>
                    <ul>
                        <li>의류 전체</li>
                        <li>재킷 &amp; 베스트</li>
                        <li>탑 &amp; 티셔츠</li>
                        <li>팬츠 &amp; 타이츠</li>
                        <li>양말</li>
                        <li>셋업</li>
                    </ul>
                </li>
                <li>
                    <ul>
                        <li>최신글</li>
                    </ul>
                </li>
            </ul>
        </StyledDetailMenu>
    );
};
const fadeIn = keyframes`
  from {
    opacity: 0;
    //display: none;
  }
  to {
    opacity: 1;
    //display: flex;
  }
`;
const fadeOut = keyframes`
  from {
    opacity: 1;
    //display: flex;
  }
  to {
    opacity: 0;
    //display: none;
  }
`;
const StyledDetailMenu = styled.div<{ isShow: boolean }>`
    //transition: opacity .15s ease-in-out, -webkit-transform .3s ease-in-out;
    display: none;
    ${({ isShow }) =>
        isShow
            ? css`
                  display: flex;
                  opacity: 1;
              `
            : css`
                  display: none;
                  opacity: 0;
              `};
    animation-name: ${({ isShow }) => isShow && fadeIn};
    animation-duration: 0.3s;
    //animation-iteration-count: 0;

    width: 100%;
    height: 372px;
    background-color: #341cff;
    border-bottom: 1px solid #fff;
    padding: 42px 0 0 238px;
    position: absolute;
    border-top: 1px solid #fff;
    top: 70px;
    left: 0;

    & > ul {
        display: flex;
        width: auto;
        height: auto;
    }

    & ul {
        margin: 0;
        padding: 0;
        list-style: none;
    }

    & > ul > li > ul {
        display: flex;
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
        flex-direction: column;
        margin-right: 60px;
    }

    & > ul > li > ul > li {
        font-family: PretendardSemiBold;
        font-size: 18px;
        color: #ffffff;
        margin-bottom: 18px;
        cursor: pointer;
        transition: transform 0.3s ease-in-out, -webkit-transform 0.3s ease-in-out;
        &:hover {
            transform: translate(3px, 0);
            -ms-transform: translate(3px, 0);
            -webkit-transform: translate(3px, 0);
        }
    }
`;
