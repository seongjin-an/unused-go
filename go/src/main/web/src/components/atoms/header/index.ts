export { BaseTopMenu } from './BaseTopMenu';
export { DetailTopMenu } from './DetailTopMenu';
export { UserName } from './func/UserName';
export { FuncButton } from './func/FuncButton';
