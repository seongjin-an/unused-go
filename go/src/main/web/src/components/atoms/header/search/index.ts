export { CloseSearchButton } from './CloseSearchButton';
export { RecentItem } from './RecentItem';
export { SearchButton } from './SearchButton';
export { SearchInput } from './SearchInput';
