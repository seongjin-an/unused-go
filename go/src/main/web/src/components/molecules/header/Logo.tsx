import React from "react";
import styled from "styled-components";
import logoIcon from '../../../static/image/dark/common/topBar/img/img_logo.png'

export const Logo: React.FC = () => {
    return (
        <StyledLogoWrapper>
            <StyledLogoIcon/>
        </StyledLogoWrapper>
    )
}
const StyledLogoWrapper = styled.div`
  flex: 1.4;
  display: flex;
  align-items: center;
  justify-content: center;
`
const StyledLogoIcon = styled.div`
  width: 128px;
  height: 43px;
  background: url(${logoIcon}) no-repeat;
`