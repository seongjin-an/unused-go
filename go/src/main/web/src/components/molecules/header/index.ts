export {Logo} from './Logo'
export {TopMenus} from './TopMenus'