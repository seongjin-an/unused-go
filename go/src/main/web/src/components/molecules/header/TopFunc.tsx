import React from 'react';
import styled from 'styled-components';
import { FuncButton, UserName } from '../../atoms/header';
interface IProps {
    handleOpenSearch: () => void;
}
export const TopFunc: React.FC<IProps> = ({ handleOpenSearch }) => {
    return (
        <div>
            <StyledTopFunc>
                <UserName />
                <FuncButton buttonType={'search'} callback={handleOpenSearch} />
                <FuncButton buttonType={'basket'} />
            </StyledTopFunc>
        </div>
    );
};

const StyledTopFunc = styled.div`
    display: flex;
    -webkit-box-align: center;
    align-items: center;
    justify-content: center;
    height: 100%;
`;
