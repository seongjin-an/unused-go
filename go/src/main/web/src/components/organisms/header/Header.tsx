import React from "react"
import styled from "@emotion/styled";
import {Logo, TopMenus} from "../../molecules/header";
import {TopFunc} from "../../molecules/header/TopFunc";

export const Header: React.FC = () => {
    return (
        <StyledTopBar>
            <Logo/>
            <StyledMenuWrapper>
                <TopMenus/>
                <TopFunc/>
            </StyledMenuWrapper>
        </StyledTopBar>
    )
}

const StyledMenuWrapper = styled.div`
  flex: 10;
  display: flex;
  justify-content: space-between;
  height: 100%;
`

const StyledTopBar = styled.section`
  //display: -webkit-box;
  //display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  //-ms-flex-pack: justify;
  //justify-content: space-between;
  -webkit-box-align: center;
  //-ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 70px;
  padding: 0;
  background-color: #21248D;
  border-bottom: 1px solid #fff;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
`