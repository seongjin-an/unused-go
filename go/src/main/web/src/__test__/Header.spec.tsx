import {configure, mount, render, shallow} from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import {Header} from "../components/organisms/header/Header";
import React from "react";

describe('header test~', () => {
    beforeEach(() => {
        console.log('=====moking test=====')

        configure({adapter: new Adapter()})
    })
    afterEach(() => {
        console.log('=====done!=====')
    })
    it('should match snapshot Header', () => {
        const component = mount(<Header/>)
        expect(component).toMatchSnapshot()
    })
})