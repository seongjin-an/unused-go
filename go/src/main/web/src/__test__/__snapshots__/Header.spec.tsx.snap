// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header test~ should match snapshot Header 1`] = `
<Header>
  <styled.section>
    <section
      className="sc-ksdxgE hoWsOj"
    >
      <Logo>
        <styled.div>
          <div
            className="sc-bdvvtL ewWqu"
          >
            <styled.div>
              <div
                className="sc-gsDKAQ ixANYz"
              />
            </styled.div>
          </div>
        </styled.div>
      </Logo>
      <styled.div>
        <div
          className="sc-bqiRlB eTOdwa"
        >
          <TopMenus>
            <styled.div
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="sc-gKclnd ouheE"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <BaseTopMenu>
                  <styled.ul>
                    <ul
                      className="sc-dkPtRN iwzmHV"
                    >
                      <li>
                        나의 중고시새
                      </li>
                      <li>
                        운동용품
                      </li>
                      <li>
                        의류
                      </li>
                      <li>
                        커뮤니티
                      </li>
                    </ul>
                  </styled.ul>
                </BaseTopMenu>
                <DetailTopMenu
                  isShow={false}
                >
                  <styled.div
                    isShow={false}
                  >
                    <div
                      className="sc-hKwDye hVXzjj"
                    >
                      <ul>
                        <li>
                          <ul>
                            <li>
                              중고시세 한눈에보기
                            </li>
                          </ul>
                        </li>
                        <li>
                          <ul>
                            <li>
                              운동용품 전체
                            </li>
                            <li>
                              아령
                            </li>
                            <li>
                              바벨
                            </li>
                            <li>
                              원판/플레이트
                            </li>
                            <li>
                              벨트
                            </li>
                            <li>
                              악력기
                            </li>
                            <li>
                              밴드/짐볼
                            </li>
                            <li>
                              스트랩/보호대
                            </li>
                          </ul>
                        </li>
                        <li>
                          <ul>
                            <li>
                              의류 전체
                            </li>
                            <li>
                              재킷 & 베스트
                            </li>
                            <li>
                              탑 & 티셔츠
                            </li>
                            <li>
                              팬츠 & 타이츠
                            </li>
                            <li>
                              양말
                            </li>
                            <li>
                              셋업
                            </li>
                          </ul>
                        </li>
                        <li>
                          <ul>
                            <li>
                              최신글
                            </li>
                          </ul>
                        </li>
                      </ul>
                    </div>
                  </styled.div>
                </DetailTopMenu>
              </div>
            </styled.div>
          </TopMenus>
          <TopFunc
            handleOpenSearch={[Function]}
          >
            <div>
              <styled.div>
                <div
                  className="sc-iCfMLu klAKLs"
                >
                  <UserName>
                    <styled.div>
                      <div
                        className="sc-eCImPb hJLDrh"
                      >
                        안김안
                        <span>
                          님
                        </span>
                      </div>
                    </styled.div>
                  </UserName>
                  <FuncButton
                    buttonType="search"
                    callback={[Function]}
                  >
                    <styled.button
                      buttonType="search"
                      onClick={[Function]}
                    >
                      <button
                        className="sc-jRQBWg cLvFvO"
                        onClick={[Function]}
                      />
                    </styled.button>
                  </FuncButton>
                  <FuncButton
                    buttonType="basket"
                  >
                    <styled.button
                      buttonType="basket"
                    >
                      <button
                        className="sc-jRQBWg cLvFvO"
                      />
                    </styled.button>
                  </FuncButton>
                </div>
              </styled.div>
            </div>
          </TopFunc>
        </div>
      </styled.div>
      <TopSearch
        handleCloseSearch={[Function]}
        isSearchShow={false}
      >
        <styled.div
          isSearchShow={false}
        >
          <div
            className="sc-egiyK fZCaFO"
          >
            <CloseSearchButton
              handleCloseSearch={[Function]}
            >
              <styled.button
                onClick={[Function]}
              >
                <button
                  className="sc-furwcr hLvxrx"
                  onClick={[Function]}
                />
              </styled.button>
            </CloseSearchButton>
            <styled.div>
              <div
                className="sc-crHmcD dwlxTp"
              >
                <SearchInput>
                  <styled.input
                    type="text"
                  >
                    <input
                      className="sc-kDTinF TWBvG"
                      type="text"
                    />
                  </styled.input>
                </SearchInput>
                <SearchButton>
                  <styled.button>
                    <button
                      className="sc-jrQzAO gofgDO"
                    />
                  </styled.button>
                </SearchButton>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-iqseJM djymIp"
              >
                <RecentItem>
                  <styled.div>
                    <div
                      className="sc-pVTFL fwENdU"
                    >
                      <div
                        className="img"
                      />
                      <div
                        className="name"
                      >
                        아령
                      </div>
                    </div>
                  </styled.div>
                </RecentItem>
                <RecentItem>
                  <styled.div>
                    <div
                      className="sc-pVTFL fwENdU"
                    >
                      <div
                        className="img"
                      />
                      <div
                        className="name"
                      >
                        아령
                      </div>
                    </div>
                  </styled.div>
                </RecentItem>
                <RecentItem>
                  <styled.div>
                    <div
                      className="sc-pVTFL fwENdU"
                    >
                      <div
                        className="img"
                      />
                      <div
                        className="name"
                      >
                        아령
                      </div>
                    </div>
                  </styled.div>
                </RecentItem>
                <RecentItem>
                  <styled.div>
                    <div
                      className="sc-pVTFL fwENdU"
                    >
                      <div
                        className="img"
                      />
                      <div
                        className="name"
                      >
                        아령
                      </div>
                    </div>
                  </styled.div>
                </RecentItem>
                <RecentItem>
                  <styled.div>
                    <div
                      className="sc-pVTFL fwENdU"
                    >
                      <div
                        className="img"
                      />
                      <div
                        className="name"
                      >
                        아령
                      </div>
                    </div>
                  </styled.div>
                </RecentItem>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </TopSearch>
    </section>
  </styled.section>
</Header>
`;
